contracts/btc-finality/src/contract.rs

```diff
@@ -249,18 +245,16 @@ fn handle_end_block(
         res = res.add_events(events);
     }
 
-    // On an epoch boundary, send rewards for distribution to Babylon Genesis
+    // On a reward distribution boundary, send rewards for distribution to Babylon Genesis
     if env.block.height > 0 && env.block.height % cfg.reward_interval == 0 {
-        let rewards = TOTAL_PENDING_REWARDS.load(deps.storage)?;
-        if rewards.u128() > 0 {
-            let (fp_rewards, wasm_msg) = send_rewards_msg(deps, rewards.u128(), &cfg)?;
+        update_rewards_dist_for_interval(deps, &env)?;
```

Comment:
It seems that within the if statement of L249, we write into `REWARDS`, read `REWARDS` for constructing reward message, and then clear `REWARDS` right after.

Is it possible to simply avoid this `REWARDS` storage i.e.,

- calls `update_rewards_dist_for_interval` which reads voting power table and contract balance, and returns `fp_rewards` as `Vec<RewardInfo>`
- If `fp_rewards` is empty, then return
- calls `send_rewards_msg` which takes `Vec<RewardInfo>` as input, and returns `wasm_msg`
- adds `wasm_msg` to `res`

This gets rid of most of the DB write operations and thus much cheaper/simpler

Reply:
Very good point. REWARDS store is legacy and not needed in the current approach. Fixed in [869d53d](https://github.com/babylonlabs-io/cosmos-bsn-contracts/pull/291/commits/869d53daff3f15ed168ff9f5c23e5c5cdbbf632d)